python/kwiver/sprokit/adapters/adapter_data_set.cxx

```diff
@@ -0,0 +1,296 @@
+  ADS_ADD_OBJECT(py::str, std::string)
+  ADS_ADD_OBJECT(kwiver::vital::image_container, std::shared_ptr<kwiver::vital::image_container>)
+  ADS_ADD_OBJECT(kwiver::vital::descriptor_set, std::shared_ptr<kwiver::vital::descriptor_set>)
+  ADS_ADD_OBJECT(kwiver::vital::detected_object_set, std::shared_ptr<kwiver::vital::detected_object_set>)
+  ADS_ADD_OBJECT(kwiver::vital::track_set, std::shared_ptr<kwiver::vital::track_set>)
+  ADS_ADD_OBJECT(kwiver::vital::feature_track_set, std::shared_ptr<kwiver::vital::feature_track_set>)
+  ADS_ADD_OBJECT(kwiver::vital::object_track_set, std::shared_ptr<kwiver::vital::object_track_set>)
+  ADS_ADD_OBJECT(std::vector<double>, std::shared_ptr<std::vector<double>>)
+  ADS_ADD_OBJECT(std::vector<std::string>, std::shared_ptr<std::vector<std::string>>)
+  ADS_ADD_OBJECT(std::vector<unsigned char>, std::shared_ptr<std::vector<unsigned char>>)
+  ADS_ADD_OBJECT(kwiver::vital::bounding_box_d, kwiver::vital::bounding_box_d)
+  ADS_ADD_OBJECT(kwiver::vital::timestamp, kwiver::vital::timestamp)
+  ADS_ADD_OBJECT(kwiver::vital::geo_polygon, kwiver::vital::geo_polygon)
+  ADS_ADD_OBJECT(kwiver::vital::f2f_homography, kwiver::vital::f2f_homography)
+
+  #undef ADS_ADD_OBJECT
+
+  throw py::type_error("Unable to add object to adapter data set");
+}
+
+// Take data of an unknown type from a port and return. Can't return as an "any" object,
+// so need to cast
+py::object get_port_data_correct_type(adapter_data_set &self, ::sprokit::process::port_t const& port)
+{
+  kwiver::vital::any const any = self.get_port_data<kwiver::vital::any>(port);
+
+  #define ADS_GET_OBJECT(TYPE) \
+  if (any.is_type<TYPE>()) \
+  { \
+    return py::cast(kwiver::vital::any_cast<TYPE>(any)); \
+  }
+
+  ADS_GET_OBJECT(int)
+  ADS_GET_OBJECT(float)
+  ADS_GET_OBJECT(std::string)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::image_container>)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::descriptor_set>)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::detected_object_set>)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::track_set>)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::feature_track_set>)
+  ADS_GET_OBJECT(std::shared_ptr<kwiver::vital::object_track_set>)
+  ADS_GET_OBJECT(std::shared_ptr<std::vector<double>>)
+  ADS_GET_OBJECT(std::shared_ptr<std::vector<std::string>>)
+  ADS_GET_OBJECT(std::shared_ptr<std::vector<unsigned char>>)
+  ADS_GET_OBJECT(kwiver::vital::bounding_box_d)
+  ADS_GET_OBJECT(kwiver::vital::timestamp)
+  ADS_GET_OBJECT(kwiver::vital::geo_polygon)
+  ADS_GET_OBJECT(kwiver::vital::f2f_homography)
+
+  #undef ADS_GET_OBJECT
+
+  std::string msg("Unable to convert object found at adapter data set port: ");
+  msg += port;
+  msg += ". Data is of type: ";
+  msg += any.type_name();
+  throw py::type_error(msg);
+}
+
+}
+}
+}
+
+
+PYBIND11_MODULE(adapter_data_set, m)
+{
+  py::bind_vector<std::vector<unsigned char>, std::shared_ptr<std::vector<unsigned char>>>(m, "VectorUChar", py::module_local(true));
```

Comment:
Comment here describing these would be good.

Reply:
Added in a0322134e3f4a072f38fcba0b4da0bab9d227a23. Is that clear?

Reply:
Yes, this is helpful.